design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,254 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**                                      | **Description**                                              |
+| --------------------------------------------- | ------------------------------------------------------------ |
+| Epic                                          | https://github.com/cyberark/conjur/issues/1892               |
+| Ruby CLI                                      | https://github.com/cyberark/conjur-cli                       |
+| Go CLI                                        | https://github.com/cyberark/conjur-cli-go                    |
+| Python CLI                                    | https://github.com/cyberark/conjur-api-python3               |
+| [FPM](https://github.com/jordansissel/fpm)    | For building packages for multiple platforms (RPM, DEB, OS X) |
+| [Debify](https://github.com/conjurinc/debify) | Build a package from a Ruby gem                              |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
```

Comment:
are we planning to provide tooling to deploy Dockerless Conjur too? if not, this is kind of misleading.

also, for what it's worth there's a lot of value in having a Dockerless CLI option regardless of the deployment mode of Conjur - it might be worth speaking to that here, instead?

Reply:
> also, for what it's worth there's a lot of value in having a Dockerless CLI option regardless of the deployment mode of Conjur - it might be worth speaking to that here, instead?

Completely agreed here - the current CLI can be improved in many aspects and would add value independent of Conjur deployment style

Reply:
> are we planning to provide tooling to deploy Dockerless Conjur too? if not, this is kind of misleading.

Yes. The whole project's goal is to make everything dockerless 😈 

> also, for what it's worth there's a lot of value in having a Dockerless CLI option regardless of the deployment mode of Conjur - it might be worth speaking to that here, instead?

We basically already have that in Go CLI and Python CLI. But do you mean there is a lot of value in delivering a CLI that we official support?

Reply:
wow - I didn't realize we were working on a Dockerless Conjur OSS too. is there a design doc where I can read up more about that?

Reply:
From what I understand, even though Conjur is a component of DAP, we will not be _officially_ supporting a Dockerless Conjur. You are correct that it is misleading and therefore dropping any mention of Dockerless _Conjur_.